backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/CsvApi.java

```diff
@@ -50,10 +52,52 @@ private static String discardMetadata(Request request, Response response) {
   }
 
   static String mergeMetadata(Request request, Response response) {
-    SchemaMetadata schema = Emx2.fromRowList(getRowList(request));
-    getSchema(request).migrate(schema);
-    response.status(200);
-    return "add/update metadata success";
+    try {
+      SchemaMetadata schema = Emx2.fromRowList(getRowList(request));
+      getSchema(request).migrate(schema);
+      response.status(200);
+      return "add/update metadata success";
+    } catch (MolgenisException me) {
+      // the upload might actual table data instead of metadata
+      // but, we don't know which table because there is no filename (no multipart form)
+      // therefore, use findTableByColumns() to figure out where the data must go
+      Iterable<Row> rows = getRowList(request);
+      Row firstRow = rows.iterator().next();
+      Table t = findTableByColumns(firstRow.getColumnNames(), getSchema(request));
```

Comment:
Try to avoid single letter variables, it's more readable for others if you write out names.

Reply:
Also, what if there is a different error. That is not handled now.